src/common/components/repository-row/index.js

```diff
@@ -73,11 +75,28 @@ class RepositoryRow extends Component {
   }
 
   onRegisterClick(repositoryUrl) {
-    const { snap } = this.props;
+    const { snap, dispatch, fullName } = this.props;
     const repository = parseGitHubRepoUrl(repositoryUrl);
     const { snapName } = this.state;
     const triggerBuilds = (!!snap.snapcraft_data);
-    this.props.dispatch(registerName(repository, snapName, triggerBuilds));
+    let clientValidationError = false;
+
+    if (/^-|-$/.test(snapName)) {
+      clientValidationError = {
+        message: 'Sorry the name can\'t start or end with a hyphen.'
+      };
+    } else if (!/[a-z0-9-]/.test(snapName)) {
+      // XXX we can't get here on any browser that supports pattern validation
```

Comment:
Safari will allow you to submit even with a pattern attr ... but I can't test this just yet because of #222